game-core/src/main/java/games/strategy/engine/message/unifiedmessenger/EndPoint.java

```diff
@@ -91,19 +91,16 @@ public int getLocalImplementorCount() {
    * @return we have no more implementors.
    */
   boolean removeImplementor(final Object implementor) {
-    synchronized (implementorsMutex) {
-      if (!implementors.remove(implementor)) {
-        throw new IllegalStateException(
-            "Not removed, impl:" + implementor + " have " + implementors);
-      }
-      return implementors.isEmpty();
+    if (!implementors.remove(implementor)) {
+      throw new IllegalStateException("Not removed, impl:" + implementor + " have " + implementors);
```

Comment:
Side-comment (as this is existing behavior):

I've questioned a bit why the original code throws on this case.
Perhaps because we're trying to detect a bug where we did not remove an implementor and we got the name wrong?

If that is the motivation, then adding a remove listener is probably a better way to go, or having a 'removeAll' hook maybe.

Reply:
removeAll maybe, but I don't think all use cases always remove everything.
No idea why the exception is there either.